# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Parent  99f5ff3c628b69fd723f2200db72e750301f52be
Feature: Minimise to tray for Linux: Change GTK call from gtk_window_present() to gtk_window_present_with_time().

diff --git a/mailnews/base/src/nsMessengerUnixIntegration.cpp b/mailnews/base/src/nsMessengerUnixIntegration.cpp
--- a/mailnews/base/src/nsMessengerUnixIntegration.cpp
+++ b/mailnews/base/src/nsMessengerUnixIntegration.cpp
@@ -66,17 +66,17 @@ static void activateWindow(mozIDOMWindow
   if (!widget) {
     fprintf(stderr, "Betterbird: Activating window via Focus() only (2)\n");
     return;
   }
 
   // Raise the window, no good if it appears in the background.
   // KDE works with just the `SetFocus()`, `gtk_window_present` was added for Gnome.
   void* gtkWidget = widget->GetNativeData(NS_NATIVE_SHELLWIDGET);
-  if (gtkWidget) gtk_window_present(GTK_WINDOW(gtkWidget));
+  if (gtkWidget) gtk_window_present_with_time(GTK_WINDOW(gtkWidget), (guint32)(g_get_monotonic_time() / 1000));
   widget->SetFocus(nsIWidget::Raise::Yes, mozilla::dom::CallerType::System);
 
   // Now move the window to where it was. We need to find it first in the
   // list of windows.
   if (!gtkWidget) {
     fprintf(stderr, "Betterbird: Activating window via Focus() only (3)\n");
     return;
   }
@@ -150,17 +150,17 @@ static void activated(AppIndicator* self
     nsCOMPtr<nsIWidget> widget;
     window->GetMainWidget(getter_AddRefs(widget));
     if (!widget) {
       continue;
     }
     // Raise the window, no good if it appears in the background.
     // KDE works with just the `SetFocus()`, `gtk_window_present` was added for Gnome.
     void* gtkWidget = widget->GetNativeData(NS_NATIVE_SHELLWIDGET);
-    if (gtkWidget) gtk_window_present(GTK_WINDOW(gtkWidget));
+    if (gtkWidget) gtk_window_present_with_time(GTK_WINDOW(gtkWidget), (guint32)(g_get_monotonic_time() / 1000));
     widget->SetFocus(nsIWidget::Raise::Yes, mozilla::dom::CallerType::System);
 
     WinDims dims = sHiddenWindowsDims[i];
     // The following isn't working well, so use native calls.
     // Last parameter: Bits: eRepaint = 1; eDelayResize = 2;
     // window->SetPositionAndSize(dims.x, dims.y, dims.w, dims.h, 1);
     if (gtkWidget) {
       gtk_window_move(GTK_WINDOW(gtkWidget), dims.x, dims.y);
